File: Examples/ClassDiagram/Swift/Podcast.swift
import Foundation

/// One episode of a podcast.
public final class Podcast: MediaItem {
    public let host: String
    public let episodeNumber: Int

    public init(title: String, duration: TimeInterval, host: String, episodeNumber: Int) {
        self.host = host
        self.episodeNumber = episodeNumber
        super.init(title: title, duration: duration, genre: .spokenWord)
    }

    public override func play() {
        print("🎙 Episode \(episodeNumber): \(title)")
    }
}

File: Examples/ClassDiagram/Python/podcast.py
from media_item import MediaItem
from playable import Genre


class Podcast(MediaItem):
    """One episode of a podcast."""

    def __init__(self, title: str, duration: float, host: str, episodeNumber: int):
        super().__init__(title, duration, Genre.SPOKEN_WORD)
        self.host: str = host
        self.episodeNumber: int = episodeNumber

    def play(self) -> None:
        print(f"🎙 Episode {self.episodeNumber}: {self.title}")
